types/views/views.go

```diff
@@ -170,6 +170,23 @@ func SliceContains[T comparable](v Slice[T], e T) bool {
 	return false
 }
 
+// SliceEquals reports whether a and b contain the same elements, regardless of order.
+func SliceEquals[T comparable](a, b Slice[T]) bool {
+	if a.Len() != b.Len() {
+		return false
+	}
+	aHasValue := make(map[T]bool)
```

Comment:
seems like we should try our hardest to match index-for-index between the two slices first before allocating and see how far we get. Then we only need to this allocating slow path for the remainder bit that wasn't in order.

that is, for two slices a and b both of length 10, maybe the first 5 elements of each are equal, but the last 5 are mixed up.  then we only need to do this slow path for the last 5.